termd-build-driver/src/main/java/org/jboss/pnc/termdbuilddriver/commands/TermdCommandInvoker.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.termdbuilddriver.commands;
+
+import org.jboss.pnc.termdbuilddriver.websockets.TermdTerminalConnection;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.ByteBuffer;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.function.Consumer;
+
+public class TermdCommandInvoker {
+
+    TermdTerminalConnection termdTerminalConnection;
+
+    public TermdCommandInvoker(URI serverBaseUri) {
+        this.termdTerminalConnection = new TermdTerminalConnection(serverBaseUri);
+    }
+
+    public CompletableFuture<String> performCommand(String command, Optional<Consumer<String>> logConsumer) {
+        return CompletableFuture.supplyAsync(() -> {
+            StringBuilder logs = new StringBuilder();
+            CountDownLatch numberOfCommandsReceived = new CountDownLatch(2);
+            try {
+                termdTerminalConnection.connect();
+
+                String data = "{\"action\":\"read\",\"data\":\"" + command + "\\r\\n\"}";
+                termdTerminalConnection.sendAsBinary(ByteBuffer.wrap(data.getBytes()));
+                if(logConsumer.isPresent()) {
+                    termdTerminalConnection.addLineConsumer(logConsumer.get());
+                }
+                termdTerminalConnection.addLineConsumer(logLine -> {
+                    logs.append(logLine).append("\n");
+                    if(logLine.contains("% ")) {
+                        numberOfCommandsReceived.countDown();
```

Comment:
Can you use events instead of parsing response log. See Client.connectStatusListenerClient & WEB_SOCKET_LISTENER_PATH = "/socket/process-status-updates"